jupyter/muehle-gui.ipynb

```diff
@@ -257,36 +455,259 @@
+   "metadata": {},
+   "source": [
+    "### handleGame\n",
+    "(TODO: Erklärung)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def handleGame(self, x, y):\n",
+    "    if self.winner is not None:\n",
+    "        logger.warning('Game has ended!')\n",
+    "        return\n",
     "\n",
-    "    # player 1\n",
+    "    logger.info(f'({x}, {y})')\n",
     "\n",
-    "    x = STASH_STARTING_POINT_X + PLAYER_PIECE_RADIUS\n",
-    "    y = STASH_STARTING_POINT_Y + CANVAS_PADDING + PLAYER_PIECE_RADIUS\n",
+    "    phase = playerPhase(self.state, self.player)\n",
+    "    logger.info(f'player phase: {phase}')\n",
     "\n",
-    "    for i in range(stash_p1):\n",
-    "        x += 2 * PLAYER_PIECE_RADIUS + STASH_STONES_SPACING\n",
-    "        draw_stone(canvas[1], (x, y), PLAYER_1)\n",
+    "    stone = getClickedStone(x, y)\n",
     "\n",
-    "    # player 2\n",
+    "    if stone is None:\n",
+    "        logger.warning('No stone was clicked!')\n",
+    "        return\n",
     "\n",
-    "    x = STASH_STARTING_POINT_X + PLAYER_PIECE_RADIUS\n",
-    "    y = STASH_STARTING_POINT_Y + CANVAS_PADDING + 3 * PLAYER_PIECE_RADIUS + STASH_STONES_SPACING\n",
+    "    if self.hasMillToPound:\n",
+    "        self.poundingMill(stone)\n",
+    "    else:\n",
+    "        if self.selectedStone is not None:\n",
+    "            self.moveStoneTo(stone, canJump = phase == 3)\n",
+    "        else:\n",
+    "            if phase == 1:\n",
+    "                logger.info('HI')\n",
+    "                self.playerPhaseOne(stone)\n",
+    "            else:\n",
+    "                if phase == 2 or phase == 3:\n",
+    "                    self.selectStone(stone)\n",
     "\n",
-    "    for i in range(stash_p2):\n",
-    "        x += 2 * PLAYER_PIECE_RADIUS + STASH_STONES_SPACING\n",
-    "        draw_stone(canvas[1], (x, y), PLAYER_2)\n",
-    "    "
+    "    if self.checkIfFinished():\n",
+    "        self.updateText()\n",
+    "            \n",
+    "GameState.handleGame = handleGame\n",
+    "del handleGame"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### togglePlayer\n",
+    "(TODO: Erklärung)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def togglePlayer(self):\n",
+    "    self.player = PLAYER_1 if self.player == PLAYER_2 else PLAYER_2\n",
+    "        \n",
+    "GameState.togglePlayer = togglePlayer\n",
+    "del togglePlayer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### playerPhaseOne\n",
+    "(TODO: Erklärung)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def playerPhaseOne(self, stone):\n",
+    "    if self.state[1][stone[0]][stone[1]] != NO_PLAYER:\n",
```

Comment:
Vielleicht eine `getPlayerAt()` Methode schreiben?

Reply:
hab ich in utils gemacht ✔️

Reply:
✔️